src/com/jwetherell/algorithms/graph/EdmondsKarp.java

```diff
@@ -0,0 +1,100 @@
+package com.jwetherell.algorithms.graph;
+
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+/**
+ * In computer science, the Edmonds–Karp algorithm is an
+ * implementation of the Ford–Fulkerson method for computing
+ * the maximum flow in a flow network in O(V*E^2) time.
+ * <p>
+ * https://en.wikipedia.org/wiki/Edmonds%E2%80%93Karp_algorithm
+ *
+ * @author Mateusz Cianciara <e.cianciara@gmail.com>
+ */
+public class EdmondsKarp {
+    private long[][] flow; //max flow beetween i and j verticles
+    private long[][] capacity; // edge capacity
+    private int[] P; //parent
+    private boolean[] visited; //just for checking if visited
+    private int n, m; //s - source, t - target
+
+    public EdmondsKarp(int NumOfVerticles, int NumOfEdges) {
+        this.n = NumOfVerticles;
+        this.m = NumOfEdges;
+        flow = new long[n][n];
+        capacity = new long[n][n];
+        P = new int[n];
+        visited = new boolean[n];
+    }
+
+    public void AddEdge(int from, int to, long capacity) {
+        assert capacity >= 0;
+        this.capacity[from][to] += capacity;
```

Comment:
What is from or to are larger than the number of elements of capacity.
e.g.
capacity[5][5] but 'from' or 'to' >= 5 ?